### PR TITLE
COSA-1099: Adds OmbuLabs authentication to the project

### LINKED ISSUE #19: Start using ombu_labs-auth gem
We want to stop using basic HTTP authentication and switch to our official authentication gem.

### PR DESCRIPTION
**Pull request by torresga**
This PR adds the OmbuLabs `ombu_labs-auth` gem to librarian. I also updated the models to include the model we need to get the authentication to work. I added a sign out link to the navigation. I also updated the tests as appropriate.

Screen Shot 2023-05-17 at 11 59 42 AM

https://ombulabs.atlassian.net/browse/COSA-1099

### CODE REVIEW

**Comment by torresga** [comment]
@cleicar This can be QAed now - I paired with Ariel on this, so he already had a chance to review.

**Review by arielj — APPROVED** [review]
looks good, ready to QA

**Comment by arielj** [comment]
Actually I need to reach out to dev tooling to setup the staging environment to be able to QA it properly, not ready to QA yet

**Comment by FionaDL** [comment]
@arielj did you talk to dev-tooling about this?

**Comment by arielj** [comment]
@FionaDL yes, I was waiting for the QA of the other PR, I'll send you the info in private

**Comment by FionaDL** [comment]
@arielj @torresga I tried some QA, but the app is not working, I tried to run the migrations in staging for Heroku, but am getting this error:
PG::UndefinedTable: ERROR:  relation "users" does not exist

**Comment by arielj** [comment]
@FionaDL this is now fixed in staging, I also updated the PR to use the gem from rubygems instead of github

I'm running the task that populates the list of articles now

this is ready for QA again

**Comment by arielj** [comment]
https://librarian-staging.herokuapp.com staging link

File: Gemfile [hunk 1]
```diff
@@ -12,6 +12,7 @@ gem "mechanize"
 gem "sitemap-parser", github: "fastruby/sitemap-parser",
                       branch: "fixes/recurse-to-level-n"
 gem "ombu_labs-shortener"
+gem "ombu_labs-auth", "1.0.0"
```
Comment (etagwerker) [inline Gemfile]: @arielj Do we want to use fixed versions here? This will need to be changed if we release 1.0.x and might be impractical. Thoughts?

**Review by etagwerker — COMMENTED** [review]
@arielj I can do QA for this. Could you please review my comment?

File: Gemfile (hunk 1 above)
Comment (arielj) [reply Gemfile]: for some reason it was installing `0.1.0` for me, I had to specify 1.0.0

I can remove it now (it's in the lock file anyway)

Comment (arielj) [reply Gemfile]: done :+1:

**Pull request merged**